root/recording/edit_form.tt

```diff
@@ -24,9 +24,15 @@
       [%- form_row_name_with_guesscase(r, { guessfeat => 1 }) -%]
       <div id="artist-credit-editor"></div>
       [%- form_row_text_long(r, 'comment', add_colon(l('Disambiguation'))) -%]
-      [%- IF !form.used_by_tracks || form.field('length').has_errors;
-            form_row_text_long(r, 'length', l('Length:'));
-          END -%]
+      [%- IF !form.used_by_tracks || form.field('length').has_errors -%]
+        [%- form_row_text_long(r, 'length', l('Length:')) -%]
+      [%- ELSE -%]
+        [% WRAPPER form_row %]
+          <label>[% l('Length:') %]</label>
+          [% l('{length_info|Automatically derived} from the associated track lengths.',
```

Comment:
The only thing weird to me is that it doesn't show the length that it derived anywhere, even though we know what it is and now have this "Length:" row. So I think we should display it, possibly with this message in parentheses next to it (though that causes things to wrap on my screen, so I had to remove "Automatically" to fit it). You might have a better idea.

```suggestion
          [% l('{recording_length} ({length_info|derived} from the associated track lengths)',
                recording_length => format_length(form.field('length').value),
                length_info => doc_link('Recording') ) %]
```

Reply:
Sounds like a good idea, done.